src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -358,6 +362,149 @@ public synchronized void setProjectFactory(BranchProjectFactory<P, R> projectFac
     }
 
     /**
+     * Offers direct access to set the configurable list of branch sources <strong>while</strong> preserving
+     * branch source id associations for sources that are otherwise unmodified
+     * @param sources
+     * @throws IOException
+     */
+    public void setSourcesList(List<BranchSource> sources) throws IOException {
```

Comment:
If somebody wants to build an alternative to org folders but retaining multibranch, they need this method to handle legacy navigator implementations